proto/regen/bond/v1alpha1/tx.proto

```diff
@@ -0,0 +1,84 @@
+syntax = "proto3";
+package regen.bond.v1alpha1;
+
+import "gogoproto/gogo.proto";
+import "google/protobuf/timestamp.proto";
+
+option go_package = "github.com/regen-network/regen-ledger/x/bond";
+
+// Msg defines the Msg service.
+service Msg {
+
+  // IssueBond issues a new bond
+  rpc IssueBond(MsgIssueBond) returns (MsgIssueBondResponse);
+
+  // SellBond sells a bond to a new owner
+  rpc SellBond(MsgSellBond) returns (MsgSellBondResponse);
+}
+
+// MsgIssueBond is the Msg/IssueBond request type.
+message MsgIssueBond {
+
+  // emission_denom is the unique id of emission
+  string emission_denom = 1;
+
+  // bond name
+  string name = 2;
+
+  // account address of current holder of the bond
+  string holder = 3;
+
+  // face value
+  string face_value = 4;
+
+  // currency of the face value
+  string face_currency = 5;
+
+  // date of issuance
+  google.protobuf.Timestamp issuance_date = 6 [(gogoproto.stdtime) = true];
+
+  // date of maturity
+  google.protobuf.Timestamp maturity_date = 7 [(gogoproto.stdtime) = true];
+
+  // coupon rate
+  string coupon_rate = 8;
+
+  // coupon frequency
+  string coupon_frequency = 9;
+
+  // project code
+  string project = 10;
+
+  // metadata encoded in base64
+  string metadata = 11;
+}
+
+// MsgIssueBondResponse is the Msg/IssueBond response type.
+message MsgIssueBondResponse {
+
+  // id is the unique id of the bond
+  string id = 1;
+}
+
+// MsgIssueBond is the Msg/IssueBond request type.
+message MsgSellBond {
+
+  // id is the unique id of the bond to sell
+  string id = 1;
+
+  // buyer of the bond
+  string buyer = 2;
+
+  // amount of bond to sell
+  string amount = 3;
+
+  // holder of the bond
+  string holder = 4;
+}
```

Comment:
I'm not entirely clear on how this will work without an order of some kind being stored on chain. A seller would need to create an order and a buyer would need to be able to fulfill that order.

Reply:
yeah, or kind of a multisig: both seller and buyer need to sign the message.

Spec is missing to validate this design.